src/libstd/io/stdio.rs

```diff
@@ -872,6 +872,91 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that
+/// a prompt is printed before capturing input.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input,
+/// or when input is empty without newline
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ").expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn prompt(prompt: &str) -> Result<String> {
+    stdout().write(prompt.as_bytes()).expect("failed to write to stdout");
+    Ok(input()?)
+}
+
+/// Constructs a new handle to the standard input of the current
+/// process, locks this handle and reads a line of input,
+/// returning a `String` containing the input. For automatic
+/// prompt handling, see the [`prompt`] method.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// # Note
+///
+/// This function is not the recommended solution to get user input, it should
+/// only be used for simple actions. If you require more explicit control over
+/// getting user input, see the [`Stdin::read_line`] method.
+///
+/// # Panics
+///
+/// This function currently panics if it does not receive input
+/// or when input is empty (end of file / `ctrl + d`).
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(input)]
+/// use std::io::input;
+///
+/// fn main() {
+///     print!("Enter name: ");
+///
+///     let name = input().expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "input",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn input() -> Result<String> {
+    let mut input = String::new();
+    stdout().flush().expect("failed to flush stdout");
+    match stdin().read_line(&mut input).expect("failed to read from stdin") {
+        0 => panic!("unexpected end of file"),
+        _ => Ok(String::from(input.trim_end_matches(&['\n', '\r'][..]))),
+    }
+}
```

Comment:
```suggestion
    let mut input = String::new();
    stdout().flush().expect("failed to flush stdout");
    match stdin().read_line(&mut input).expect("failed to read from stdin") {
        0 => panic!("unexpected end of file"),
        _ => Ok(String::from(input.trim_end_matches(&['\n', '\r'][..]))),
    }
}
```
There are quite a few functions panicking here, I don't think any of it does not need to be propagated, we should let the user `unwrap` or solve it. I think we should update the docs and example for this.

@QuietMisdreavus does the docs need to mention what kind of error will be propagated? It have `ErrorKind` but there are quite a few options in `ErrorKind`.

Reply:
I'm not an active team member any more, so cc @rust-lang/docs to be sure.

For a matter of personal opinion, i'm a fan of mentioning at least *some* potential error cases in the docs, if not all of them. It helps to be explicit about what's possible.